fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java

```diff
@@ -568,10 +570,28 @@ public Statistics visitLike(Like like, EstimationContext context) {
                     "col stats not found. slot=%s in %s",
                     like.left().toSql(), like.toSql());
             ColumnStatisticBuilder colBuilder = new ColumnStatisticBuilder(origin);
-            colBuilder.setNdv(origin.ndv * DEFAULT_LIKE_COMPARISON_SELECTIVITY).setNumNulls(0);
+            double notNullNdv = getNotNullSelectivity(origin, origin.ndv * DEFAULT_LIKE_COMPARISON_SELECTIVITY);
+            colBuilder.setNdv(origin.ndv * DEFAULT_LIKE_COMPARISON_SELECTIVITY)
+                    .setCount(notNullNdv * context.statistics.getRowCount()).setNumNulls(0);
             statsBuilder.putColumnStatistics(like.left(), colBuilder.build());
             context.addKeyIfSlot(like.left());
         }
         return statsBuilder.build();
     }
+
+    private double getNotNullSelectivity(ColumnStatistic stats, double origSel) {
+        double rowCount = stats.count;
+        double numNulls = stats.numNulls;
+        double ndv = stats.ndv;
+        if (numNulls > rowCount - ndv) {
+            numNulls = rowCount - ndv > 0 ? rowCount - ndv : 0;
```

Comment:
"numNulls = rowCount-ndv" this is not true. 
consider col values (1, 1, 1, null)
rowcount=4, ndv=1
then we get numNulls = 4-1=3, but it is 1

Reply:
current impl may lead to the inconsistent status between rowCount and ndv, so it will go into this handling logic unexpectly. I will comment it first and refine it in future.